File: js-adaptor/src/System/IO/FileInfo.js
/**
 * @compiler Bridge.NET 17.10.1
 */
Bridge.assembly("unity-script-converter", function($asm, globals) {
    "use strict";

    Bridge.define("System.IO.FileInfo", {
        props: {
            Exists: {
                get: function() {
                    try {
                        this._fs.accessSync(this._wxPath);
                        return true;
                    } catch (e) {
                        console.log(e);
                        return false;
                    }
                }
            },
            Length: {
                get: function() {
                    const stat = this._fs.statSync(this._wxPath);
                    if (stat) {
                        return stat.size;
                    } else {
                        return undefined;
                    }
                }
            },
            LastWriteTimeUtc: {
                get: function() {
                    throw new System.Exception("not impl");
                },
                set: function(value) {
                    throw new System.Exception("not impl");
                }
            },
            FullName: {
                get: function() {
                    return this._wxPath;
                }
            }
        },
        ctors: {
            ctor: function(path) {
                this.$initialize();
                if (path.startsWith('/')) {
                    path = path.substring(1, path.length);
                }
                if (path.startsWith('usr')) {
                    path = path.substring(3, path.length);
                }
                this._wxPath = `${wx.env.USER_DATA_PATH}/` + path;
                console.log("FileInfo Created, Path:", this._wxPath);
                this._fs = wx.getFileSystemManager();
            }
        },
        methods: {
            Create: function() {
                try {
                    this._fs.unlinkSync(this._wxPath);
                } catch (e) {
                    console.log('FileInfo Create unlink old file', e);
                }
                return new System.IO.FileStream.$ctor1(this);
            }, 
            Open: function(mode) {
                return new System.IO.FileStream.$ctor1(this);
            }
        }
    });
});

File: unity-plugin/script-export/sdk/System~/IO/FileInfo.cs
namespace System.IO {

    [Bridge.External]
    public class FileInfo {

        public FileInfo(string path) {
            throw new Exception("not impl");
        }

        public bool Exists {
            get {
                throw new Exception("not impl");
            }
        }

        public int Length {
            get {
                throw new Exception("not impl");
            }
        }

        public DateTime LastWriteTimeUtc {
            get {
                throw new Exception("not impl");
            }
            set {
                throw new Exception("not impl");
            }
        }

        public FileStream Open(FileMode mode) {
            throw new Exception("not impl");
        }
        
        public FileStream Create() {
            throw new Exception("not impl");
        }

        public string FullName {
            get {
                throw new Exception("not impl");
            }
        }

        public StreamWriter CreateText() {
            throw new Exception("not impl");
        }

        public StreamWriter AppendText() {
            throw new Exception("not impl");
        }
        
    }
}